src/native/managed/cdacreader/src/Contracts/Metadata_1.cs

```diff
@@ -0,0 +1,385 @@
+                DwFlags = _target.Read<uint>(Address + (ulong)_type.Fields[nameof(Metadata_1.MethodTableFlags.DwFlags)].Offset),
+                DwFlags2 = _target.Read<uint>(Address + (ulong)_type.Fields[nameof(Metadata_1.MethodTableFlags.DwFlags)].Offset),
+                BaseSize = _target.Read<uint>(Address + (ulong)_type.Fields[nameof(Metadata_1.MethodTableFlags.BaseSize)].Offset),
+            };
+            _methodTableFlags = flags;
+        }
+        return _methodTableFlags.Value;
+    }
+
+    internal Metadata_1.MethodTableFlags Flags => EnsureFlags();
+
+    internal TargetPointer EEClassOrCanonMT => _target.ReadPointer(Address + (ulong)_type.Fields[nameof(EEClassOrCanonMT)].Offset);
+    internal TargetPointer EEClass => Metadata_1.GetEEClassOrCanonMTBits(EEClassOrCanonMT) == Metadata_1.EEClassOrCanonMTBits.EEClass ? EEClassOrCanonMT : throw new InvalidOperationException("not an EEClass");
+    internal TargetPointer CanonMT
+    {
+        get
+        {
+            if (Metadata_1.GetEEClassOrCanonMTBits(EEClassOrCanonMT) == Metadata_1.EEClassOrCanonMTBits.CanonMT)
+            {
+                return new TargetPointer((ulong)EEClassOrCanonMT & ~(ulong)Metadata_1.EEClassOrCanonMTBits.Mask);
+            }
+            else
+            {
+                throw new InvalidOperationException("not a canonical method table");
+            }
+        }
+    }
+}
+
+internal struct UntrustedEEClass_1
+{
+    public readonly Target _target;
+    private readonly Target.TypeInfo _type;
+
+    internal TargetPointer Address { get; init; }
+
+    internal UntrustedEEClass_1(Target target, TargetPointer eeClassPointer)
+    {
+        _target = target;
+        Address = eeClassPointer;
+        _type = target.GetTypeInfo(DataType.EEClass);
+    }
+
+    internal TargetPointer MethodTable => _target.ReadPointer(Address + (ulong)_type.Fields[nameof(MethodTable)].Offset);
+}
+
+
+internal struct MethodTable_1
+{
+    internal Metadata_1.MethodTableFlags Flags { get; }
+    internal ushort NumInterfaces { get; }
+    internal ushort NumVirtuals { get; }
+    internal TargetPointer ParentMethodTable { get; }
+    internal TargetPointer Module { get; }
+    internal TargetPointer EEClassOrCanonMT { get; }
+    internal MethodTable_1(Data.MethodTable data)
+    {
+        Flags = new Metadata_1.MethodTableFlags
+        {
+            DwFlags = data.DwFlags,
+            DwFlags2 = data.DwFlags2,
+            BaseSize = data.BaseSize,
+        };
+        NumInterfaces = data.NumInterfaces;
+        NumVirtuals = data.NumVirtuals;
+        EEClassOrCanonMT = data.EEClassOrCanonMT;
+        Module = data.Module;
+        ParentMethodTable = data.ParentMethodTable;
+    }
+}
+
+internal struct EEClass_1
+{
+    internal TargetPointer MethodTable { get; }
+    internal ushort NumMethods { get; }
+    internal ushort NumNonVirtualSlots { get; }
+    internal uint TypeDefTypeAttributes { get; }
+    internal EEClass_1(Data.EEClass eeClassData)
```

Comment:
I think this is leftovers from before I changed the public surface of the contract to be in terms of `MethodTableHandle`.  At that point `EEClass_1` leaked out since `GetClassData` was public